pkg/appmanager/resourceConfig.go

```diff
@@ -1534,16 +2001,18 @@ func (rc *ResourceConfig) AddRuleToPolicy(
 func (rc *ResourceConfig) DeleteRuleFromPolicy(
 	policyName string,
 	rule *Rule,
+	mergedRulesMap map[string]map[string]mergedRuleEntry,
 ) {
 	// We currently have at most 1 policy, 'forwarding'
 	policy := rc.FindPolicy("forwarding")
 	if nil != policy {
 		for i, r := range policy.Rules {
 			if r.Name == rule.Name && r.FullURI == rule.FullURI {
 				// Remove old rule
+				unmerged := rc.UnmergeRule(rule.Name, mergedRulesMap)
 				if len(policy.Rules) == 1 {
 					rc.RemovePolicy(*policy)
```

Comment:
Only remove the policy if len() is 1 AND !unmerged. If the rule we are deleting was merged with another, and we unmerge and delete the initial rule, we are left with the unmerged rule (which shouldn't be deleted), but we will delete the policy anyway because there is 1 rule.

Reply:
Fixed